components/sync15/src/error.rs

```diff
@@ -112,14 +106,16 @@ pub enum ErrorKind {
     #[fail(display = "Unexpected server behavior during batch upload: {}", _0)]
     ServerBatchProblem(&'static str),
 
-    #[fail(display = "Setup state machine cycle detected")]
-    SetupStateCycleError,
+    #[fail(
```

Comment:
Hmm, I wonder if we should expose these. I guess they're already going to try again later, so it's probably fine.